test/nodetool/test_ring.py

```diff
@@ -0,0 +1,158 @@
+#
+# Copyright 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+from typing import NamedTuple
+import pytest
+import subprocess
+from rest_api_mock import expected_request
+
+
+class Host(NamedTuple):
+    dc: str
+    rack: str
+    endpoint: str
+    load: float
+    ownership: float
+
+
+def map_to_json(mapper, mapped_type=None):
+    json = []
+    for key, value in mapper.items():
+        if mapped_type is None:
+            json.append({'key': key, 'value': value})
+        else:
+            json.append({'key': key, 'value': mapped_type(value)})
+    return json
+
+
+def format_stat(width, address, rack, status, state, load, owns, token):
+    return f"{address:<{width}}  {rack:<12}{status:<7}{state:<8}{load:<16}{owns:<20}{token:<44}\n"
+
+
+def format_size(v):
+    units = {
+        1 << 40: 'TB',
+        1 << 30: 'GB',
+        1 << 20: 'MB',
+        1 << 10: 'KB',
+    }
+    for n, unit in units.items():
+        if v > n:
+            d = v / n
+            return f'{d:.2f} {unit}'
+    return f'{v} bytes'
+
+
+def test_ring(nodetool):
+    keyspace = 'ks'
+    host = Host('dc0', 'rack0', '127.0.0.1', 6414780.0, 1.0)
+    token_to_endpoint = {
+        "-9217327499541836964": host.endpoint,
+        "9066719992055809912": host.endpoint,
+        "50927788561116407": host.endpoint,
+    }
+    endpoint_to_ownership = {
+        host.endpoint: host.ownership,
+    }
+    status_to_endpoints = {
+        'live': ['127.0.0.1'],
+        'down': [],
+    }
+    state_to_endpoints = {
+        'joining': [],
+        'leaving': [],
+        'moving': [],
+    }
```

Comment:
Can you please send a follow-up, parameterizeing this test so that we have at least one of each of these states (also down state above).

Reply:
sure. will do.

Reply:
tracked at https://github.com/scylladb/scylladb/issues/17401